### PR TITLE
Adapt design to styleguide

### LINKED ISSUE #48: Apply Nimiq Styleguide
Apply new Nimiq styling, can use https://github.com/nimiq/nimiq-style.

### PR DESCRIPTION
**Pull request by sisou**
Resolves #48.

Use @nimiq/nimiq-style in Accounts Manager.

Also includes additions of `buttonText` and `disabled` props to the `Success.vue` component to enable `SignTransactionSuccess.vue` to use it.

Screenshots:
screenshot from 2018-11-15 10-32-31
screenshot from 2018-11-15 10-34-12
screenshot from 2018-11-15 10-34-46
screenshot from 2018-11-15 10-36-00

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -7,6 +7,9 @@
     <script src="https://cdn.nimiq-testnet.com/web-offline.js"></script>
     <link rel="icon" href="<%= BASE_URL %>favicon.ico">
     <title>Nimiq Accounts Manager</title>
+
+    <link href="https://fonts.googleapis.com/css?family=Muli:400,600,700" rel="stylesheet">
+    <link href="https://fonts.googleapis.com/css?family=Fira+Mono&text=0123456789ABCDEFGHJKLMNPQRSTUVXY" rel="stylesheet">
```
Comment (svub) [inline public/index.html]: Ah, yes, we need a monospace. Has info on this been added to the style guide?

File: src/components/CheckoutDetails.vue [hunk 2]
```diff
@@ -62,27 +64,30 @@ export default class CheckoutDetails extends Vue {
...
     .data {
-        font-size: 2rem;
-        line-height: 1.3;
-        opacity: 0.7;
-        padding: 1rem 4rem;
+        margin: 1rem 4rem;
+    }
+
+    .transaction-section {
+        background: white;
```
Comment (svub) [inline src/components/CheckoutDetails.vue]: should constants be in place for all colors?

Comment (svub) [reply src/components/CheckoutDetails.vue]: Say, I'd assume the Nimiq style lib would provide constants such as colors to be used in CSS. Same issue in other vue files. It's a pain to change something when hard-coded in each file. :/

File: src/views/AddAccountSuccess.vue [hunk 3]
```diff
@@ -1,29 +1,29 @@
...
+    <div class="container pad-bottom">
+        <SmallPage>
             <PageHeader>Your account is ready</PageHeader>
-            <div class="page-body">
-                <div class="success-box">
-                    <h2>Cool!</h2>
-                    <p>Your new account has been added to your wallet.</p>
+            <PageBody>
+                <div class="success-box nq-icon trumpet nq-bg-green">
+                    <h2 class="nq-h2">Awesome!</h2>
```
Comment (svub) [inline src/views/AddAccountSuccess.vue]: Should we also ride the "Awesome" wave? Alternatives: "Perfect!", "Nice!", or maybe "Delightful!", "Sweet!"

File: src/components/CheckoutDetails.vue (hunk 2 above)
Comment (svub) [reply src/components/CheckoutDetails.vue]: Actually, further down I saw 
```
            background: var(--nimiq-blue);
            background-image: var(--nimiq-blue-bg);
```
so I'd suggest to extend this approach to really have all colors in constants, including `white`, `#f0f0f0`, `#fafafa`, and `#e5e5e5`.

File: src/views/SignupTypeSelector.vue [hunk 4]
```diff
@@ -60,57 +62,29 @@ export default class SignupTypeSelector extends Vue {
...
-        margin: 0;
+        cursor: pointer;
     }
 
-    button.purple-background {
-        background: #724ceb;
-        color: white;
-        border: none;
+    .ledger-button {
+        border: solid 2px #e5e5e5;
```
Comment (svub) [inline src/views/SignupTypeSelector.vue]: color

File: src/views/SignupTypeSelector.vue [hunk 5]
```diff
@@ -60,57 +62,29 @@ export default class SignupTypeSelector extends Vue {
...
+        color: inherit;
     }
 
-    button.keyguard-icon::after,
-    button.ledger-icon::after {
+    button.keyguard-button::after,
+    button.ledger-button::after {
         content: '';
         display: block;
         background: #fafafa;
```
Comment (svub) [inline src/views/SignupTypeSelector.vue]: color

File: src/components/CheckoutDetails.vue (hunk 2 above)
Comment (svub) [reply src/components/CheckoutDetails.vue]: (added note "color" to some of the spots)

File: public/index.html (hunk 1 above)
Comment (sisou) [reply public/index.html]: Yes, `Fira Mono` is mentioned in the style guide.

File: src/App.vue [hunk 6]
```diff
@@ -74,35 +67,34 @@ export default class App extends Vue {
...
         box-sizing: border-box;
     }
 
     .logo {
         height: 3.625rem;
         box-sizing: border-box;
-        padding-left: 6rem;
+        /* padding-left: 6rem; */
         font-size: 2.125rem;
         font-weight: 600;
```
Comment (nibhar) [inline src/App.vue]: Later you changed all numerical values for font-weight. Why not here?

File: src/App.vue [hunk 7]
```diff
@@ -134,7 +126,6 @@ export default class App extends Vue {
     .loading-animation + h2 {
         margin-top: 4rem;
         padding-bottom: 8rem;
-        color: white;
         text-transform: uppercase;
         font-size: 1.75rem;
         font-weight: 600;
```
Comment (nibhar) [inline src/App.vue]: font-weight numerical value again.

File: src/views/AddAccountSuccess.vue (hunk 3 above)
Comment (sisou) [reply src/views/AddAccountSuccess.vue]: Wording is by no means final. Mostly just a placeholder until we figure out a proper wording revision process.

File: src/components/Success.vue [hunk 8]
```diff
@@ -1,23 +1,30 @@
...
-        <div class="icon-checkmark-circle"></div>
-        <h1 v-html="parsedText"></h1>
+    <div class="success nq-bg-green">
+        <div class="nq-icon checkmark-circle"></div>
+        <h1 v-html="parsedText" class="nq-h1"></h1>
         <div style="flex-grow: 1;"></div>
-        <button @click="$emit('continue')">Back to {{ appName }}</button>
+
+        <button v-if="buttonText" @click="$emit('continue')" class="nq-button green inverse" :disabled="disabled">{{ buttonText }}</button>
+        <button v-else @click="$emit('continue')" class="nq-button green inverse" :disabled="disabled">Back to {{ appName }}</button>
```
Comment (nibhar) [inline src/components/Success.vue]: Could be done as getter instead of v-if.

File: src/views/CheckoutTransmission.vue [hunk 9]
```diff
@@ -55,65 +58,33 @@ export default class CheckoutTransmission extends Vue {
...
+        animation: color-shift 1s;
     }
 
-    @keyframes enter {
+    @keyframes color-shift {
         0% {
-            background: #6843df;
+            background: var(--nimiq-blue);
+            background-image: var(--nimiq-blue-bg);
             max-height: 160px;
```
Comment (nibhar) [inline src/views/CheckoutTransmission.vue]: i'd like 20rem better here.

File: src/App.vue (hunk 6 above)
Comment (sisou) [reply src/App.vue]: What's wrong with numerical values though? `normal` is 400, `bold` is 700. There is no name for 600.

File: src/views/SignupTypeSelector.vue (hunk 4 above)
Comment (sisou) [reply src/views/SignupTypeSelector.vue]: This is a one-off color not specified by the theme. It's only in the designs for the accounts manager so far.

File: src/components/Success.vue (hunk 8 above)
Comment (sisou) [reply src/components/Success.vue]: If you mean that the button text could be a getter, then yes, I also though that, until I noticed that then we would have to escape `appName` manually.

File: src/App.vue (hunk 6 above)
Comment (nibhar) [reply src/App.vue]: nothing wrong with them i just saw you changed it in another file.

File: src/components/Success.vue (hunk 8 above)
Comment (nibhar) [reply src/components/Success.vue]: would
```
get getButtonText() {
  return (buttonText) ? buttonText : 'Back to " + appName;
}
```
not get escaped when rendered as `{{ getButtonText }}`

Comment (sisou) [reply src/components/Success.vue]: You are of course right xD

File: src/App.vue (hunk 6 above)
Comment (svub) [reply src/App.vue]: Just to keep in mind, adding a font-weight should be about 20kb extra to download.

File: src/views/SignupTypeSelector.vue (hunk 4 above)
Comment (svub) [reply src/views/SignupTypeSelector.vue]: Should we do that??

Comment (svub) [reply src/views/SignupTypeSelector.vue]: Just quickly searched for `color: ` and found as an example `SignupSuccess.vue`. It has twice hard-code the same color in just this one file. I think we should only have a fixed set of constants representing the colors in the style guide. I can do a fork+PR replacing all the colors. Let me know what you think is best to do.

Comment (sisou) [reply src/views/SignupTypeSelector.vue]: Just replaced the last hardcoded color value here: https://github.com/nimiq/accounts/pull/55/commits/9bfc1f3e530cfda01e99221893bedb77855a92f1. The other places you found must have been in outdated code.

**Pull request merged**